x/shield/keeper/purchase.go

```diff
@@ -98,21 +101,31 @@ func (k Keeper) purchaseShield(ctx sdk.Context, poolID uint64, shield sdk.Coins,
 
 	// Check pool shield limit.
 	poolParams := k.GetPoolParams(ctx)
+	protectionEndTime := ctx.BlockTime().Add(poolParams.ProtectionPeriod)
 	maxShield := sdk.MinInt(pool.ShieldLimit, totalCollateral.Sub(totalWithdrawing).ToDec().Mul(poolParams.PoolShieldLimit).TruncateInt())
 	if shieldAmt.Add(pool.Shield).GT(maxShield) {
 		return types.Purchase{}, types.ErrPoolShieldExceedsLimit
 	}
 
-	// Send service fees to the shield module account and update service fees.
-	if err := k.DepositNativeServiceFees(ctx, serviceFees, purchaser); err != nil {
-		return types.Purchase{}, err
+	// get next purchase ID and set purchase ID after that
+	purchaseID := k.GetNextPurchaseID(ctx)
+	k.SetNextPurchaseID(ctx, purchaseID+1)
+	if !serviceFees.IsZero() {
```

Comment:
Can we panic if both service and staking fees are non-zero? Or we can use a single coin but add a boolean.

Reply:
I'm pretty sure they're calculated on the public function to be non-zero, but we can add another check to be sure.